### ISSUE TITLE
Inserting multiple items ignores defaults for missing values, but sets NULL instead

### ISSUE THREAD

**Issue by elhigu** [tgriesser/knex#662]
Documentation says:
```
// Normalizes for empty keys on multi-row insert:
knex('coords').insert([{x: 20}, {y: 30},  {x: 10, y: 20}])
Outputs:
insert into `coords` (`x`, `y`) values (20, NULL), (NULL, 30), (10, 20)
```

Would it be possible to change this behavior to actually use DEFAULT instead of NULL?

Adding implicit NULL causes e.g. not null constraint failure in case where one would like to set default value.

**Comment by bendrucker** [comment]
I think this is sensible. @tgriesser?

**Comment by demisx** [comment]
+1 Just got bitten by it again. Once the seed file has some values for fields, their defaults never kick in.

**Comment by elhigu** [comment]
Giving DEFAULT as special value seems to be supported by mariadb and postgresql... Is there any preference how this should be supported? 

If I understood correctly it did seem that default value was meant to be able to be set in database configuration (https://github.com/tgriesser/knex/blob/master/src/dialects/postgres/index.js#L20).

To me it sounds reasonable to have this as a configuration option, since this feature cannot be implemented e.g. for sqlite and I would like to keep default behaviour of knex consistent between dialect implementations (would be a bit awkward if defaults are set correctly for mysql and postgresql, but differently for sqlite).

I already hacked together simple implementation for this, if the configuration selectable value approach is fine I could start putting clean implementation together with tests and documentation.

**Comment by elhigu** [comment]
Looks like https://github.com/tgriesser/knex/pull/962 has also some discussion about multi insert.

**Comment by chrisbroome** [comment]
:+1: I was very surprised by this behavior also.

Unless there is a **very** good reason not to, my columns are always `NOT NULL` with `DEFAULT` values. The current behavior assumes that every column is nullable which is a really really bad practice for relational schema design.

**Comment by MatthewPhinney** [comment]
As far as I can tell this was not completely fixed until #1269, included in version 11.

**Comment by nik-lampe** [comment]
For me this still happens when I run the seeds programatically.

Running 
 await knex.seed.run({ specific: 'dev.js', env: 'test' }) 
sets `null` for the missing values
whereas running 
knex seed:run --specific dev.js --env test
sets `DEFAULT`
